docs update to match the new user config (#235)
transposit/docs: building/user-config.md

Each data connection in your application can either reference a production key that you have provided or expect signed-in users to provide their own key. Check the appropriate boxes under **Require users to authenticate with these data connections**. For the rest, navigate to **Deploy &gt; Production Keys** and click **Add key**.

## User settings schema

> For applications requiring user-specific input, you can create a user settings schema under **User &gt; User Configuration &gt; User settings schema**. Any settings that are marked as `Show on user settings page` will be displayed for user input, along with the required authentications, when your app is visited.

Reviewer: ```suggestion
For applications requiring user-specific configuration, you can define a user settings schema at **Users &gt; User Configuration &gt; User settings schema**. Settings marked with `Show on user settings page` will be displayed on the app's user settings page, along with any required data connector authentication.
```

2. Commit your code
3. Create a user setting of type `options`
4. Select your newly-created operation in the dropdown
5. Save your changes

## Custom display of operation results

> Transposit's automatically generated settings and authentication page will be sufficient for most use cases. If your application requires more UI, you have the option of selecting `(Advanced) This app requires a custom display for operation results`.

Reviewer: ```suggestion
If your application requires more customization than the automatically-generated user settings page provides, you can select `Advanced (custom page template)`.
```

> With a custom display, you can modify the HTML and JavaScript provided in **Code &gt; Page template** to invoke operations and display results. For complex front ends, this can be a great starting point when prototyping.

Reviewer: ```suggestion
With a custom page template, you can supply custom settings page HTML and JavaScript in **Code &gt; Page template** to invoke operations and display results. This can be a great starting point for prototyping application UI.
```

> For even more control, it is possible to host your app's frontend with a third party and use Transposit as a backend. Apps can interface with Transposit via the [JavaScript SDK](/docs/building/js-sdk) or by directly invoking [HTTP endpoints](/docs/building/endpoints).

Reviewer: ```suggestion
You can also host your app's frontend with a third party service and use Transposit as a backend. Apps can interface with Transposit via the [JavaScript SDK](/docs/building/js-sdk) or by directly invoking [HTTP endpoints](/docs/building/endpoints).
```
